cuda/base/cusparse_bindings.hpp

```diff
@@ -45,9 +45,29 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 namespace gko {
+/**
+ * @brief The device specific kernels.
+ *
+ * \ingroup kernels
+ */
 namespace kernels {
+/**
+ * @brief The CUDA namespace.
+ *
+ * \ingroup cuda
+ */
 namespace cuda {
+/**
+ * @brief The CUSPARSE namespace.
+ *
+ * \ingroup cusparse
+ */
 namespace cusparse {
+/**
+ * @brief The detail namespace.
```

Comment:
Since all of these have a `@ingroup`, what happens if you remove the `@brief`? Does the namespace get the documentation of the group? I am not really a fan of all of that duplication (sometimes the `@brief` is not the same as in other places).

Another question if the previous one does not work, what happens if you document these only once, and remove all `@brief` for the other ones.

Reply:
Actually another question: do you even need to specify `@ingroup` multiple times after you did this once?

Reply:
Yes, I should have removed most duplications. But in the inner-most nested namespace if the parent namespace is different, you still need to add some doc otherwise it is not visible.